src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -0,0 +1,512 @@
+        {
+            // We would not have been asked to produce an example of a missing pattern if no values are missing
+            Debug.Assert(!remainingValues.IsEmpty);
+
+            // If the input is an enumeration type, see if any declared enumeration constant values are in the set.
+            // If so, that is what to report.
+            if (type is NamedTypeSymbol { TypeKind: TypeKind.Enum } e)
+            {
+                foreach (var declaredMember in e.GetMembers())
+                {
+                    if (declaredMember is FieldSymbol { IsConst: true, IsStatic: true, DeclaredAccessibility: Accessibility.Public } field &&
+                        field.GetConstantValue(ConstantFieldsInProgress.Empty, false) is ConstantValue constantValue &&
+                        remainingValues.Any(BinaryOperatorKind.Equal, constantValue))
+                    {
+                        return field.ToDisplayString();
+                    }
+                }
+            }
+
+            var sample = remainingValues.Sample;
+            if (sample != null)
+                return ValueString(sample, type, requireExactType);
+
+            // IValueSet.Sample cannot produce a sample of type `nint` or `nuint` outside the range
+            // of values of `int` and `uint`. So if we get here we need to produce a pattern indicating
+            // such an out-of-range value.
+
+            var underlyingType = type.EnumUnderlyingTypeOrSelf();
+            Debug.Assert(underlyingType.IsNativeIntegerType);
+            if (underlyingType.SpecialType == SpecialType.System_IntPtr)
+            {
+                if (remainingValues.Any(BinaryOperatorKind.GreaterThan, ConstantValue.Create(int.MaxValue)))
+                    return $"> ({type.ToDisplayString()})int.MaxValue";
+
+                if (remainingValues.Any(BinaryOperatorKind.LessThan, ConstantValue.Create(int.MinValue)))
+                    return $"< ({type.ToDisplayString()})int.MinValue";
+
+                throw ExceptionUtilities.Unreachable;
+            }
+            else if (underlyingType.SpecialType == SpecialType.System_UIntPtr)
+            {
+                if (remainingValues.Any(BinaryOperatorKind.GreaterThan, ConstantValue.Create(uint.MaxValue)))
+                    return $"> ({type.ToDisplayString()})uint.MaxValue";
+
+                throw ExceptionUtilities.Unreachable;
+            }
+
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        private static string ValueString(ConstantValue value, TypeSymbol type, bool requireExactType)
+        {
+            if ((type.IsEnumType() || requireExactType || type.IsNativeIntegerType) &&
+                !(typeHasExactTypeLiteral(type) && !value.IsNull))
+            {
+                return $"({type.ToDisplayString()}){PrimitiveValueString(value, type.EnumUnderlyingTypeOrSelf())}";
```

Comment:
>PrimitiveValueString(value, type.EnumUnderlyingTypeOrSelf()) [](start = 52, length = 60)

Consider calculating this unconditionally as the return value, then conditionally adding the `$"({type.ToDisplayString()})"` cast.